ravenframework/Models/EnsembleModel.py

```diff
@@ -545,7 +568,15 @@ def submit(self,myInput,samplerType,jobHandler,**kwargs):
     ## works, we are unable to pass a member function as a job because the
     ## pp library loses track of what self is, so instead we call it from the
     ## class and pass self in as the first parameter
-
+    if  self.localJobHandler is None and self.parallelStrategy == 2:
+      # create local clone of jobhandler
+      self.localJobHandler = jobHandler.createCloneJobHandler()
+      # start the job handler
+      self.localPollingThread = InterruptibleThread(target=self.localJobHandler.startLoop)
+      self.localPollingThread.daemon = True
+      self.localPollingThread.start()
+      # register function to kill the thread at the end of the execution
+      atexit.register(self.localPollingThread.kill)
```

Comment:
Hm, pollingThread (both here and in Simulation.py) possibly should be inside JobHandler. That said, this is just mimicking how Simulation does this, so I am not sure if this needs to be fixed now.

Reply:
Also, can the localPollingThread be killed sooner, rather than waiting for the program to exit?

Reply:
@alfoa Could you address Josh's comment here.

Reply:
added in a dedicated function that is executed at the end of the step